src/Compilers/Test/Core/TestableFileSystem.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Roslyn.Utilities;
+
+namespace Roslyn.Test.Utilities
+{
+    public delegate Stream OpenFileExFunc(string filePath, FileMode mode, FileAccess access, FileShare share, int bufferSize, FileOptions options, out string normalizedFilePath);
+    public delegate Stream OpenFileFunc(string filePath, FileMode mode, FileAccess access, FileShare share);
+
+    public sealed class TestableFileSystem : ICommonCompilerFileSystem
+    {
+        public OpenFileFunc OpenFileFunc { get; set; } = delegate { throw new InvalidOperationException(); };
+        public OpenFileExFunc OpenFileExFunc { get; set; } = (string _, FileMode _, FileAccess _, FileShare _, int _, FileOptions _, out string _) => throw new InvalidOperationException();
```

Comment:
>(string _, FileMode _, FileAccess _, FileShare _, int _, FileOptions _, out string _) => [](start = 61, length = 88)

Just curious, why use lambda syntax here but `delegate { }` syntax for the other cases?